### DISCUSSION TITLE
A couple of small QoL suggestions:

### DISCUSSION THREAD

**Discussion by kekopop** [Hypfer/Valetudo#1872] [Q&A / Support] [answered]
A couple of small QoL suggestions:

1) Can't for the life of me find any way to reboot the robot from the web interface. Sure, one can always do it via SSH, but a reboot button somewhere would be nice :)

2) Would also be nice if the enabled/disabled check box icons on the timer items in the list were interactive, so you could disable/enable timers without entering the edit view:

![image](https://github.com/Hypfer/Valetudo/assets/103499578/1c2bfbc3-5f7a-4df2-9d04-c0441250d530)

**Comment by Hypfer** [comment, marked as answer]
While this isn't a question in the questions section, I guess I should somehow publicly document my stance on the reboot stuff.

Regarding 1:

The thing with a reboot button is that it is actually a terrible thing to have, because it's something that should **never** be required.
If the software doesn't work and needs to sometimes be rebooted, the right solution is to fix the software.
However, if you add a low-friction way of "solving" the issue by just pressing the reboot button, you soften the pain of the issue faced so much that it's not enough to build up the motivation to sit down and actually solve the issue at its root.

I am of course aware that other software has reboot buttons and there that can make sense. It is quite costly after all to identify and fix a problem at its root. It's very cheap to just reboot the damn thing.

Also, paying customers do not care if a solution is elegant or correct or the right thing to do for long-term sustainability.
They want their itch scratched and that also happens if they have to occasionally press reboot.
Something not working - maybe even for months - on the other hand is terrible for the end user. They will hate it, because they do not care. They just want the thing to work. And that's perfectly valid.

But(!) Valetudo doesn't target end users. While it can be used by them, the sole driving force and reason behind such technical decisions is that it is my fun hobby project where I get to do what I want to do. "What I want to do" in this case is build excellent software. This also results in happy end-users, but that's just an unintended side-effect. If it wouldn't, that wouldn't matter.

Sorry if I'm a bit preachy there but I _really need to hammer the point home_, because most of the software in the world for some reason is user-focussed and therefore people expect that all software will be. But this isn't. This is just a hobby project that happens to be free and fairly neat to use as an end-user.

Anyway, because of that, I do not want a reboot button in Valetudo. Instead, I want Valetudo to be software that never needs a reboot.



Regarding 2:

Huh, yeah. Not sure why they aren't editable in that view. Will think about it. Thx for the input.

**Reply by kekopop** [reply]
Can see that you have a very strong opinion in this question, and that I'd never be able to convince you, but there's also the manufacturer software running in the bottom. Can't imagine they share your views on software excellency. If I'm to guess, they're all about maximizing profits. 

I work at sea, leaving my robot unsupervised for long periods. Not that I've had any problems with my current robot, but I like to reboot it preventively before leaving, to give it a fresh start so to speak. If nothing else, it gives me some kind of peace of mind.

You do however have a very valid point, and yes, I do understand that this is your hobby project, don't interpret the above as an attempt to make you change your mind :)

Thanks :) Running different timers when I'm at sea or at home, being able to toggle them directly from the list would be very convenient.
